streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopicManagerTest.java

```diff
@@ -287,12 +290,124 @@ public void shouldLogWhenTopicNotFoundAndNotThrowException() {
+        final AdminClient admin = EasyMock.createNiceMock(AdminClient.class);
+        final InternalTopicManager topicManager = new InternalTopicManager(admin, new StreamsConfig(config));
+        final TopicPartitionInfo partitionInfo = new TopicPartitionInfo(0, broker1,
+                Collections.singletonList(broker1), Collections.singletonList(broker1));
+
+        final KafkaFutureImpl<TopicDescription> topicDescriptionFailFuture = new KafkaFutureImpl<>();
+        topicDescriptionFailFuture.completeExceptionally(new LeaderNotAvailableException("Leader Not Available!"));
+        final KafkaFutureImpl<TopicDescription> topicDescriptionSuccessFuture = new KafkaFutureImpl<>();
+        topicDescriptionSuccessFuture.complete(
+            new TopicDescription(topic, false, Collections.singletonList(partitionInfo), Collections.emptySet())
+        );
+
+        EasyMock.expect(admin.describeTopics(Collections.singleton(leaderNotAvailableTopic)))
+            .andReturn(new MockDescribeTopicsResult(
+                Collections.singletonMap(leaderNotAvailableTopic, topicDescriptionFailFuture)))
+            .once();
+        EasyMock.expect(admin.createTopics(Collections.emptySet()))
+            .andReturn(new MockCreateTopicsResult(Collections.emptyMap())).once();
+
+        EasyMock.expect(admin.describeTopics(Collections.singleton(leaderNotAvailableTopic)))
+            .andReturn(new MockDescribeTopicsResult(
+                Collections.singletonMap(leaderNotAvailableTopic, topicDescriptionSuccessFuture)))
+            .once();
+
+        EasyMock.replay(admin);
+
+        final InternalTopicConfig internalTopicConfig = new RepartitionTopicConfig(leaderNotAvailableTopic, Collections.emptyMap());
+        internalTopicConfig.setNumberOfPartitions(1);
+        final Map<String, InternalTopicConfig> topicConfigMap = new HashMap<>();
+        topicConfigMap.put(leaderNotAvailableTopic, internalTopicConfig);
+        topicManager.makeReady(topicConfigMap);
+
+        EasyMock.verify(admin);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenKeepsTopicLeaderNotAvailable() {
+        final String leaderNotAvailableTopic = "LeaderNotAvailableTopic";
+        final AdminClient admin = EasyMock.createNiceMock(AdminClient.class);
+        final InternalTopicManager topicManager = new InternalTopicManager(admin, new StreamsConfig(config));
+
+        final KafkaFutureImpl<TopicDescription> topicDescriptionFailFuture = new KafkaFutureImpl<>();
+        topicDescriptionFailFuture.completeExceptionally(new LeaderNotAvailableException("Leader Not Available!"));
+
+        // simulate describeTopics got LeaderNotAvailableException
+        EasyMock.expect(admin.describeTopics(Collections.singleton(leaderNotAvailableTopic)))
+            .andReturn(new MockDescribeTopicsResult(
+                Collections.singletonMap(leaderNotAvailableTopic, topicDescriptionFailFuture)))
+            .times(2);
```

Comment:
nit: we could set a final int for numRetries as:
```
            put(StreamsConfig.adminClientPrefix(StreamsConfig.RETRIES_CONFIG), numRetries);

```
and use (numRetries + 1) here to clearly indicate we are trying to go beyond the retry limit.

Reply:
good suggestion!